pkg/client/unversioned/clientcmd/api/v1/types.go

```diff
@@ -54,7 +54,10 @@ type Preferences struct {
 type Cluster struct {
 	// Server is the address of the kubernetes cluster (https://hostname:port).
 	Server string `json:"server"`
-	// APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	// DEPRECATED: APIVersion is the preferred api version for communicating with the kubernetes cluster (v1, v2, etc).
+	//  Because a cluster can run multiple API groups and potentially multiple versions of each, it no longer makes sense to specify
+	//  a single value for the cluster version. This field isn't really needed anyway, so we're deprecating it without a replacement.
+	//  It will be ignored if it is present.
```

Comment:
> What happens in this scenario if you have "experimental/v1" and "experimental/v2"?

@JanetKuo and I decided easiest is first setting wins. In the future, we could extend this to be a preference order-- so kubectl would be allowed to use anything in the list, but would use first if possible before trying second & further ones. You can't do that with a map. Also, you can't really express a map on the command line flag, and the behavior should be consistent between them.

> "must" versus "must if available"

Hm, my understanding was that the current flag is "must"? IMO if you're setting the flag, you don't want it to silently not do what you asked-- if what you asked was impossible, don't you want to know about it?